pkg/daemon/ceph/mon/daemon.go

```diff
@@ -1,162 +0,0 @@
-You may obtain a copy of the License at
-	http://www.apache.org/licenses/LICENSE-2.0
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-// Package mon provides methods for configuring and running Ceph mons.
-package mon
-
-import (
-	"fmt"
-	"net"
-	"os"
-	"strings"
-
-	"github.com/rook/rook/pkg/clusterd"
-	cephconfig "github.com/rook/rook/pkg/daemon/ceph/config"
-	"github.com/rook/rook/pkg/util"
-)
-
-const (
-	// DefaultPort is the default port Ceph mons use to communicate amongst themselves.
-	DefaultPort = 6790
-)
-
-// Config contains the necessary parameters Rook needs to know to set up a mon for a Ceph cluster.
-type Config struct {
-	Name     string
-	Cluster  *cephconfig.ClusterInfo
-	isDaemon bool
-	Port     int32
-}
-
-// FlattenMonEndpoints returns a comma-delimited string of all mons and endpoints in the form
-// <mon-name>=<mon-endpoint>
-func FlattenMonEndpoints(mons map[string]*cephconfig.MonInfo) string {
-	endpoints := []string{}
-	for _, m := range mons {
-		endpoints = append(endpoints, fmt.Sprintf("%s=%s", m.Name, m.Endpoint))
-	}
-	return strings.Join(endpoints, ",")
-}
-
-// ParseMonEndpoints parses a flattened representation of mons and endpoints in the form
-// <mon-name>=<mon-endpoint> and returns a list of Ceph mon configs.
-func ParseMonEndpoints(input string) map[string]*cephconfig.MonInfo {
-	logger.Infof("parsing mon endpoints: %s", input)
-	mons := map[string]*cephconfig.MonInfo{}
-	rawMons := strings.Split(input, ",")
-	for _, rawMon := range rawMons {
-		parts := strings.Split(rawMon, "=")
-		if len(parts) != 2 {
-			logger.Warningf("ignoring invalid monitor %s", rawMon)
-			continue
-		}
-		mons[parts[0]] = &cephconfig.MonInfo{Name: parts[0], Endpoint: parts[1]}
-	}
-	return mons
-}
-
-// Run generates configuration files for a Ceph mon and then runs a mon daemon with the config
-func Run(context *clusterd.Context, config *Config) error {
-	configFile, monDataDir, err := generateConfigFiles(context, config)
-	if err != nil {
-		return fmt.Errorf("failed to generate mon config files. %+v", err)
-	}
-
-	err = startMon(context, config, configFile, monDataDir)
-	if err != nil {
-		return fmt.Errorf("failed to run mon. %+v", err)
-	}
-
-	return err
-}
-
-func generateConfigFiles(context *clusterd.Context, config *Config) (string, string, error) {
-
-	// write the keyring to disk
-	if err := writeMonKeyring(context, config.Cluster, config.Name); err != nil {
-		return "", "", err
-	}
-
-	// write the config file to disk
-	confFilePath, err := generateConnectionConfigFile(context, config.Cluster, GetMonRunDirPath(context.ConfigDir, config.Name),
-		"admin", getMonKeyringPath(context.ConfigDir, config.Name))
```

Comment:
Since the last note disappeared, the mon settings in config file being generated wasn't being loaded by `ceph-mon`. As it turned out the `"admin`" user spec here was causing the config file to be generated as 

```ini
[global]
... stuff ...

[admin.client]
... mon settings ...
```

Changing the user here to `"mon.<mon-name>"` causes the `[admin.client]` section to be correctly sectioned `[mon.<name>]`.